src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -315,6 +315,29 @@ export class FileService implements IFileService {
 		});
 	}
 
+	public setContentsAndResolve(resource: uri, absolutePath: string, value: string, addBom: boolean, encodingToWrite: string, options: { encoding?: string; mode?: number; flag?: string; }): TPromise<IFileStat> {
```

Comment:
Some suggestions:
* make the method `private`
* rename it to `doSetContentsAndResolve`
* move it below the method `updateContent`
* remove the `encoding` from the options, it is not being used from the caller and it is also dangerous because in order to support all encodings, we need to use the `encoding.encode` method instead